presto-hive/src/main/java/com/facebook/presto/hive/parquet/RichColumnDescriptor.java

```diff
@@ -13,25 +13,33 @@
  */
 package com.facebook.presto.hive.parquet;
 
+import com.facebook.presto.hive.HiveType;
 import parquet.column.ColumnDescriptor;
 import parquet.schema.PrimitiveType;
 
+import javax.annotation.concurrent.Immutable;
+
 import static parquet.schema.Type.Repetition.OPTIONAL;
 
 // extension of parquet's ColumnDescriptor. Exposes full Primitive type information
+@Immutable
 public class RichColumnDescriptor
         extends ColumnDescriptor
 {
     private final PrimitiveType primitiveType;
     private final boolean required;
 
+    private final HiveType hiveType;
+
     public RichColumnDescriptor(
             ColumnDescriptor descriptor,
-            PrimitiveType primitiveType)
+            PrimitiveType primitiveType,
+            HiveType hiveType)
     {
         super(descriptor.getPath(), primitiveType.getPrimitiveTypeName(), primitiveType.getTypeLength(), descriptor.getMaxRepetitionLevel(), descriptor.getMaxDefinitionLevel());
         this.primitiveType = primitiveType;
         this.required = primitiveType.getRepetition() != OPTIONAL;
+        this.hiveType = hiveType;
```

Comment:
`this.hiveType = requireNonNull(hiveType, "hiveType is null");`

It would be nice if you add non-null checks to other fields too (preferably in a separate commit).

Reply:
i think we need hiveType to be optional looking at couple of places where richColumnDescriptor is initialized https://github.com/prestodb/presto/pull/11118/files#diff-0f09736fc6c9cfc0691776d4365d409bL233

Reply:
Then please update the constructor to accept `Optional< HiveType>` and add the non-null check.